tcp_check/datadog_checks/tcp_check/tcp_check.py

```diff
@@ -68,37 +69,43 @@ def __init__(self, name, init_config, instances):
                         self.CONFIGURATION_ERROR_MSG.format(self.host, 'IPv6 address', 'valid address')
                     )
             # It's a correct IP V6 address
-            self._addr = self.host
+            self._addrs = self.host
             self.socket_type = socket.AF_INET6
         else:
             self.socket_type = socket.AF_INET
             # IP will be resolved at check time
 
     @property
-    def addr(self):
-        if self._addr is None:
+    def addrs(self):
+        if self._addrs is None or self._addrs == []:
             try:
                 self.resolve_ip()
             except Exception as e:
                 self.log.error(str(e))
                 msg = "URL: {} could not be resolved".format(self.host)
                 raise CheckException(msg)
-        return self._addr
+        return self._addrs
 
     def resolve_ip(self):
```

Comment:
```suggestion
    def resolve_ips(self):
```
nit if we're updating `self._addr` to `self._addrs`